### PR TITLE
Key management routes to biome rest api

### PR DESCRIPTION
**Pull request by eloaverona**
This PR adds a POST, GET and PATCH `/biome/users/{user_id}/keys` to the key management feature in biome.

### CODE REVIEW

File: libsplinter/src/biome/key_management/database/postgres/migrations/2019-12-12-200956_create-keys/up.sql [hunk 1]
```diff
@@ -18,5 +18,6 @@ CREATE TABLE IF NOT EXISTS keys (
     encrypted_private_key TEXT NOT NULL,
     user_id               TEXT NOT NULL,
     display_name          TEXT NOT NULL,
-    PRIMARY KEY(public_key, user_id)
```
Comment (peterschwarz) [inline libsplinter/src/biome/key_management/database/postgres/migrations/2019-12-12-200956_create-keys/up.sql]: As this migration may be part of a release, this change should probably be in a separate migration.

File: libsplinter/src/rest_api/mod.rs [hunk 2]
```diff
@@ -380,6 +380,29 @@ pub fn percent_encode_filter_query(input: &str) -> String {
     percent_encoding::utf8_percent_encode(input, QUERY_ENCODE_SET).to_string()
 }
 
+pub fn get_header(header_key: &str, request: &HttpRequest) -> Result<String, RequestError> {
```
Comment (peterschwarz) [inline libsplinter/src/rest_api/mod.rs]: Would suggest naming this `require_header`, as it fails if the header is missing.

File: libsplinter/src/biome/key_management/database/postgres/migrations/2019-12-12-200956_create-keys/up.sql (hunk 1 above)
Comment (eloaverona) [reply libsplinter/src/biome/key_management/database/postgres/migrations/2019-12-12-200956_create-keys/up.sql]: Addressed

File: libsplinter/src/rest_api/mod.rs (hunk 2 above)
Comment (eloaverona) [reply libsplinter/src/rest_api/mod.rs]: Addressed

**Comment by eloaverona** [comment]
@peterschwarz I think the change to the model introduced by Shawn should be done in a separate PR, it sounds to me out of scope for this PR.

**Pull request merged**